### PR TITLE
5735 set up uat

### PR DESCRIPTION
**Pull request by pankaj-cgi**
Code changes required for UAT environment:
- Adding option to select UAT in github workflow
- Making keycloak client-id as configurable parameter.

@weskubo-cgi : Can we remove the .env.<environment> files now with application reading custom frontend properties from config.js?

### CODE REVIEW

File: backend/src/test/resources/application-dev.yaml [hunk 1]
```diff
@@ -10,6 +10,11 @@ server:
   
   
 spring:
+  security:
+    oauth2:
+      resourceserver:
+        jwt:
+          audience: MSPDIRECT-SERVICE
```
Comment (daveb-hni) [inline backend/src/test/resources/application-dev.yaml:17]: Will this be potentially be different per environment?

**Review by daveb-hni — COMMENTED** [review]
See question.

**Review by daveb-hni — COMMENTED** [review]
See question.

**Comment by weskubo-cgi** [comment]
I'd prefer to leave the env files in as the code supports both ways of configuration and they will be used during local development and will support standard builds using vite build.

File: frontend/env_config/.env.dev [hunk 2]
```diff
@@ -2,4 +2,5 @@
 VITE_APP_TITLE=HN Web (Dev)
 VITE_KEYCLOAK_URL=https://common-logon-dev.hlth.gov.bc.ca/auth/
 VITE_SITEMINDER_LOGOUT=https://logontest7.gov.bc.ca/clp-cgi/logoff.cgi?retnow=1&returl=https://mspdirect-dev.apps.silver.devops.gov.bc.ca
-VITE_SERVICE_URL=https://mspdirect-dev-api.apps.silver.devops.gov.bc.ca/
\ No newline at end of file
+VITE_SERVICE_URL=https://mspdirect-dev-api.apps.silver.devops.gov.bc.ca/
+VITE_KEYCLOAK_CLIENT_ID=MSPDIRECT-WEB
```
Comment (weskubo-cgi) [inline frontend/env_config/.env.dev:6]: This parameter also needs to be removed from .env if we are making it environment specific.

File: backend/src/test/resources/application-dev.yaml (hunk 1 above)
Comment (daveb-hni) [reply backend/src/test/resources/application-dev.yaml:17]: Confirmed that it can be different in each env.

**Review by daveb-hni — APPROVED** [review]
Confirmed changes are good.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]
I think there's just the one issue with updating .env to address.
Also, just wanted to make sure that application-dev.yaml will be updated in svn.

File: frontend/env_config/.env.dev (hunk 2 above)
Comment (pankaj-cgi) [reply frontend/env_config/.env.dev:6]: Updated

**Comment by pankaj-cgi** [comment]
> I think there's just the one issue with updating .env to address. Also, just wanted to make sure that application-dev.yaml will be updated in svn.
@weskubo-cgi Configurations for backend are saved in application.properties in ArgoCD repo. I feel we do not need to maintain application-dev.yaml file in SVN.

**Pull request merged**